### ISSUE TITLE
Save dialog with local directory not working.

### ISSUE THREAD

**Issue by osortega** [microsoft/vscode-remote-release#981]
- VSCode Version: 1.37.0-insider (8e0f348413f4f616c23a88ae30030efa85811973)
- Local OS Version: Windows 10
- Remote OS Version: Debian GNU/Linux 9 (stretch)
- Remote Extension/Connection Type: Cloud Environments

Steps to Reproduce:

1. Connect to any environment
2. Execute 'Export logs' command
3. Remote directory shows up instead of the window pop up referencing the local directory.

![Capture](https://user-images.githubusercontent.com/48293249/61321738-0707f700-a7c1-11e9-86b3-6c0279145104.PNG)
![Capture2](https://user-images.githubusercontent.com/48293249/61321747-0a02e780-a7c1-11e9-87f1-7eb9d3153fe0.PNG)

**Comment by Chuxel** [comment]
@alexr00 This is the code they are using to try to get a local file URI to use for saving. 

```TypeScript
await vscode.window.showSaveDialog({
                    defaultUri: vscode.Uri.file(os.homedir()),
                    filters: dialogFilters,
                });
```

I suspect the `defaultUri` property may be getting removed with the changes for https://github.com/microsoft/vscode-remote-release/issues/712 - is there some other way they could default the saveAs action to be local?

**Comment by Chuxel** [comment]
//cc: @kieferrm given @alexr00 is OOF.

**Comment by rebornix** [comment]
IMO #712 doesn't change `window.showSaveDialog`'s behavior as `defaultUri` is `file` scheme, which will be translated to `vscode-remote` as that resource exists in the remote environment but not the UI side. 

```
await vscode.window.showSaveDialog({
    defaultUri: vscode.Uri.file(os.homedir()),
    filters: dialogFilters,
});
```

With above code, `os.homedir()` returns the home directory of the remote environment thus current behavior is correct: opening the dialog with the remote directory.

Right now both `window.showOpenDialog` and `window.showSaveDialog` only opens the directory from the workspace. To support local directory on the UI side, we may want to 

* either support *Local* always like what we do for *File: Open...* and *File: Save As...*
* or expose `availableFileSystems` or something similar to extensions, which can be used to decide what's the default 

The second one might be foreign to extensions authors as we need to differentiate *Local on the UI side* and *Local on the Remote side*

**Comment by rebornix** [comment]
I pushed a change to always display *Local* option when extensions execute `vscode.window.showSaveDialog` and `vscode.window.showOpenDialog` API and users can make the decision where they want to the file to be saved. Since these two APIs only return `Uri`, the extensions should use `vscode.fs.writeFile` to save the content to the target `Uri`.

The dialog will read remote authority to decide the default environment where the dialog should be opened. With a sample extension I created in https://github.com/rebornix/ui-save-dialog (install visx from https://github.com/rebornix/ui-save-dialog/releases/tag/v0.1)

when working in a local workspace, VS Code will open local directory when the API is executed

![local-ext-save-dialog](https://user-images.githubusercontent.com/876920/61553269-e3d28700-aa0e-11e9-8cd5-0536dcc04032.gif)

The extension is marked as UI extension by purpose but when working in a remote workspace, VS Code still opens the remote directory by default due to the fact that the workspace is remote.

![ui-ext-save-dialog](https://user-images.githubusercontent.com/876920/61553352-14b2bc00-aa0f-11e9-9bae-4921154c76d0.gif)

If the extension is marked as Workspace extension, it will behave the same as above. The only thing to note is if extensions try to use Nodejs `os` or `fs` to specify `defaultUri`, the result will depend on `extensionKind`.
